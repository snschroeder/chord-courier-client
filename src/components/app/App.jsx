import React from 'react';
import { Route, Switch } from 'react-router-dom';
import SearchForm from '../search-form/SearchForm';
import AboutPage from '../about-page/AboutPage';
import KeySearchForm from '../search-form/KeySearchForm';
import LandingPage from '../landing-page/LandingPage';
import NotFoundPage from '../not-found-page/NotFoundPage';
import './app.css';

function App() {
  return (
    <div className="App">
      <main className="App-main">
        <Switch>
          <Route
            exact
            path="/"
            component={LandingPage}
          />
          <Route
            path="/search"
            component={SearchForm}
          />
          <Route
            path="/full-search"
            component={KeySearchForm}
          />
          <Route
            path="/about"
            component={AboutPage}
          />
          <Route
            component={NotFoundPage}
          />
        </Switch>
      </main>
    </div>
  );
}

export default App;
