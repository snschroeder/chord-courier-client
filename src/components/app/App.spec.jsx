/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
