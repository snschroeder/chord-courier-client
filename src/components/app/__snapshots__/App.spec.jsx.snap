// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="App-main"
  >
    <nav
      className="nav-bar"
    >
      <ul
        className="link-list"
      >
        <a
          href="/search"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li
            className="nav-link"
          >
            Progression Randomizer
          </li>
        </a>
        <a
          href="/full-search"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li
            className="nav-link"
          >
            Chords by Key
          </li>
        </a>
        <a
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li
            className="nav-link"
          >
            About
          </li>
        </a>
      </ul>
    </nav>
    <section
      className="app-description"
    >
      <div
        className="hero-image"
      >
        <div
          className="hero-text"
        >
          <section>
            <h1
              className="light-header-h1"
            >
              Chord Courier
            </h1>
            <h2
              className="light-header-h2"
            >
              Delivering fresh chords
            </h2>
          </section>
          <h3
            className="hero-h3"
          >
            your catalyst for musical inspiration
          </h3>
          <h4
            className="hero-h4"
          >
            Explore chord progressions, make them your own
          </h4>
          <a
            href="/search"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <h5
              className="hero-h5"
            >
              Get started
            </h5>
          </a>
        </div>
      </div>
    </section>
  </main>
</div>
`;
